### DISCUSSION TITLE
Fix for fontawesome icons/glyphs not properly displaying on Linux desktop after installing texlive-full

### DISCUSSION THREAD

**Discussion by Vishvamitra** [FortAwesome/Font-Awesome#19283] [Show and tell]
# The problem
I installed `texlive-full` two days ago and, after some minutes I spent on editing a couple of  files with vim, I noticed that the icons on both i3bar and the terminal emulator were rendered completely wrong: some of them looked like accents, some like ligatures, some like korean characters.

# Reasearching the issue
I asked my dearest friend (aka Google Search) for a consultation on the issue and, to my astonishment, I found only a couple of pertinent reports. Then, I read this article (https://linux.debian.bugs.dist.narkive.com/ciDjKum8/bug-866114-texlive-fonts-extra-icon-errors-when-installed-alongside-fonts-font-awesome) where its author suggested that this behaviour could be caused by a conflict between fontawesome and some korean fonts.

# The solution 
Well, I have two news for you now: one is good and the other one is meh.

- Good news: you don't need to completely uninstall `texlive-fonts-extra` or, as some users had suggested, even to go as far as uninstalling `texlive-full`: these suggestions are drastical and could lead to unwanted results (in Debian Testing, for example, in its current state at the time of my writing, removing the `texlive-fonts-extra` triggers the complete removal of all LaTeX packages). So, I found a less dangerous fix, which consists in removing only the sanskrit and the korean/japanese/chinese fonts which caused the issue. Please check that the folders on my system match with yours (you can run `fc-list | sort -k 2` to get the exact location of each font installed on your system and sort the fonts by their name). Here are the fonts which you should remove in order to reenable the broken FontAwesome icons (the names ending in / are **folders**):

```
/usr/share/fonts/truetype/arphic-bkai00mp/
/usr/share/fonts/truetype/arphic-gkai00mp/
/usr/share/fonts/truetype/arphic-bsmi00lp/
/usr/share/fonts/truetype/arphic-gbsn00lp/
/usr/share/fonts/truetype/arphic/
/usr/share/fonts/truetype/baekmuk/
/usr/share/fonts/truetype/unfonts-core/
/usr/share/fonts/truetype/unfonts-extra/
/usr/share/fonts/truetype/vlgothic/
/usr/share/fonts/opentype/ebgaramond/EBGaramond08*
/usr/share/fonts/truetype/ebgaramond/EBGaramond08*
/usr/share/fonts/woff/ebgaramond/EBGaramond08*
/usr/share/fonts/opentype/ebgaramond/EBGaramondSC08-Regular.otf
/usr/share/fonts/truetype/ebgaramond/EBGaramondSC08-Regular
/usr/share/fonts/woff/ebgaramond/EBGaramondSC08-Regular.woff 
```
After the removal, it's crucial to type the `fc-cache` command in order to refresh the fonts  repository on the system.

- Meh news: by uninstalling all the previous fonts, a good 80% of fontawesome icons came back to the way they used to appear before installing `texlive-full`: the remaining ones were still broken. Unfortunately I wasn't able to track down the fonts which caused this strange behaviour, but from what I could guess these fonts are either cyrillic or ligatures/medieval fonts. If you want to take a test by yourself and you're having this issue, try to copy-paste the *battery* glyph or the *person* glyph from Font Awesome on your terminal.

Hopefully, this brief enquiry of mine will be helpful to those few individuals who will come across this rather annoying issue.
If, by any chance, you get any idea of other fonts causing this issue, please feel free to note it down below: I'm sure it will be much appreciated.

**Comment by KebabLord** [comment]
@Vishvamitra thank you man, i didn't even notice the problem was latex, huge thanks
